### ISSUE TITLE
Error when passing a limit > 100 or values for until or since

### ISSUE THREAD

**Issue by Vininski** [jcwillox/up-bank-api#2]
Beginner coder here so I couldn't fully understand all the code but it seems the date functionality in incomplete, it that right?

>trans_list = transactional.transactions(limit=101)
  File "C:\Users\Vininski\AppData\Local\Programs\Python\Python37\lib\site-packages\upbankapi\models.py", line 90, in transactions
    limit=limit, since=since, until=until, tag=tag, account_id=self.id
  File "C:\Users\Vininski\AppData\Local\Programs\Python\Python37\lib\site-packages\upbankapi\client.py", line 76, in transactions
    transactions = self.api(endpoint, params=params)["data"]
KeyError: 'data'`

`date = datetime.datetime.now()`
> trans_list = transactional.transactions(until=date)
  File "C:\Users\Vininski\AppData\Local\Programs\Python\Python37\lib\site-packages\upbankapi\models.py", line 90, in transactions
    limit=limit, since=since, until=until, tag=tag, account_id=self.id
  File "C:\Users\Vininski\AppData\Local\Programs\Python\Python37\lib\site-packages\upbankapi\client.py", line 76, in transactions
    transactions = self.api(endpoint, params=params)["data"]
KeyError: 'data'

```
year = 2020
month = 5
day = 1
hour = 1
min = 1
sec = 1
microsec = 1
tzinfo = None
date = datetime.datetime(year,month,day,hour,min,sec,microsec,tzinfo)
```
> trans_list = transactional.transactions(since=date)
  File "C:\Users\Vininski\AppData\Local\Programs\Python\Python37\lib\site-packages\upbankapi\models.py", line 90, in transactions
    limit=limit, since=since, until=until, tag=tag, account_id=self.id
  File "C:\Users\Vininski\AppData\Local\Programs\Python\Python37\lib\site-packages\upbankapi\client.py", line 76, in transactions
    transactions = self.api(endpoint, params=params)["data"]
KeyError: 'data'

Also noticed what looks to be a typo at line 69 in clients.py
```
        if until:
            params["filter[tag]"] = tag
```

**Comment by jcwillox** [comment]
Thanks this is great
The `limit` currently refers to `pagesize` which is limited to 100 items. I've yet to add support pagination.
I'll check the date implementation, that should work.
Yep, that's a typo I'll fix that.
Whenever you get `KeyError: 'data'` that likely means the Up API returned an error (probably the data wasn't formatted correctly)

**Comment by jcwillox** [comment]
This is exactly what I needed a fresh set of eyes.
I've fixed those issues you mentioned, in [0.3.0](https://github.com/jcwillox/up-bank-api/releases/tag/0.3.0), I also added support for showing the actual error message from the API.
The issue with the date-times was that they had no timezone specified, that's now been fixed and it will fetch the timezone as required.
I've also added support for pagination, which was a sizable job, but a necessary one. I believe I've converted everything correctly.
